### PR TITLE
Animal Ally | Nature Soul feats.

### PR DESCRIPTION
**Pull request by Aegonek**
Opening a draft for now to check how this works.
Implemented Nature Soul and Animal Ally feats.
Notable differences from PnP: Nature Soul grants a bonus to Perception instead of Survival. Animal Ally allows choosing any kind of animal a druid would have access to. I thought it's suitable given combat-oriented nature of the game and having a penalty of being 3 levels behind, but I'm open for debate.

I would like to playtest it a little more before merging.

### CODE REVIEW

File: TabletopTweaks/Extensions/ExtentionMethods.cs [hunk 1]
```diff
@@ -462,6 +462,28 @@ static class ExtentionMethods {
                 blueprint.AddComponent(newComponent);
             }
         }
+
+        /// <summary>
```
Comment (Aegonek) [inline TabletopTweaks/Extensions/ExtentionMethods.cs:466]: This should probably by rewritten for consistency with other ReplaceComponent methods - to replace all occurences. Maybe could be added as ReplaceFirstComponent.

File: TabletopTweaks/TabletopTweaks.csproj [hunk 2]
```diff
@@ -45,33 +45,19 @@
 
 0649</NoWarn>
   </PropertyGroup>
-  <Target Name="RemoveLib" BeforeTargets="Clean" Condition="Exists('$(SolutionDir)lib')">
```
Comment (Aegonek) [inline TabletopTweaks/TabletopTweaks.csproj:48]: Anyway to exclude file from pull request? 🤔

File: TabletopTweaks/Scripts/AfterBuild.ps1 [hunk 3]
```diff
@@ -0,0 +1,5 @@
+param (
```
Comment (Aegonek) [inline TabletopTweaks/Scripts/AfterBuild.ps1:1]: To exclude.

File: TabletopTweaks/packages.config [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="Aze.Publicise.MSBuild.Task" version="1.0.0" targetFramework="net46" developmentDependency="true" />
-  <package id="Newtonsoft.Json" version="13.0.1" targetFramework="net46" />
```
Comment (Aegonek) [inline TabletopTweaks/packages.config:4]: Change to remove.

File: .editorconfig [hunk 5]
```diff
@@ -1,8 +1,8 @@
 # Remove the line below if you want to inherit .editorconfig settings from higher directories
```
Comment (Aegonek) [inline .editorconfig:1]: Not sure what happened here.

**Comment by Aegonek** [comment]
Things seem to work just fine, fixed the mess with edits.

**Review by Vek17 — APPROVED** [review]

**Pull request merged**